Add target_type to target group binding and option to pass 'hostNetowork' option through

Ashtonian opened the pull request.

bocan commented:
Just wanted to add a +1 to this.  Without hostNetwork set to true the chart won't work with EKS using Calico networking.

Ashtonian commented:
> Just wanted to add a +1 to this. Without hostNetwork set to true the chart won't work with EKS using Calico networking.

or [CNI/SNAT](https://docs.aws.amazon.com/eks/latest/userguide/pod-networking.html)

Ashtonian commented:
updated the var docs, added target type, updated alb policy sourced from [kuberenetes-sigs](https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.2.0/docs/install/iam_policy.json)

Ashtonian commented:
I updated the chart version, the custom resource hook is no longer working for me, but it wasn't before either. I was just getting logs it wasn't able to, where as now Im not. Will keep looking.

Ashtonian commented:
[chart diff](https://github.com/aws/eks-charts/compare/v0.0.39...v0.0.53)

Ashtonian commented:
This is working and ready for merge. One remaining issue is destroying the resources doesn't clean up the TargetGroupBinding CR. I've been manually working around that via  `kdel TargetGroupBinding  {{tgb name}}-tgb -n kube-system`

mogul commented on variables.tf:
```diff
@@ -65,17 +65,24 @@ variable "aws_tags" {
+  description = "Group Binding detals for TargetGroupBindings. Expected object fields: name, backend_port, target_group_arn, target_type See https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/guide/targetgroupbinding/targetgroupbinding/ for details."
   type        = any
   default     = []
+  # default     = ""
```
```suggestion
```

mogul commented on variables.tf:
```diff
@@ -65,17 +65,24 @@ variable "aws_tags" {
 
 variable "target_groups" {
-  description = "ARNs for existing load balancers that should be added via TargetGroupBindings. See https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/guide/targetgroupbinding/targetgroupbinding/ for details"
+  description = "Group Binding detals for TargetGroupBindings. Expected object fields: name, backend_port, target_group_arn, target_type See https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/guide/targetgroupbinding/targetgroupbinding/ for details."
```
```suggestion
  description = "Group Binding details for TargetGroupBindings. Expected object fields: name, backend_port, target_group_arn, target_type See https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/guide/targetgroupbinding/targetgroupbinding/ for details."
```

mogul reviewed (approved):
Approving but left a couple review comments on stray bits. Thanks for your contribution, and your patience!

Ashtonian commented:
Thanks!

Ashtonian commented:
Can you add issues back to the repo? Or is there a different place to discuss issues? 

Currently this does not clean up CRDs as they are deployed via shell, this can be fixed via local exec destroy https://stackoverflow.com/questions/64952745/terraform-and-references-in-local-exec-destroy-provisioners

mogul commented:
I guess this fork has diverged enough from the upstream that I may as well turn on issues.

mogul commented:
Issue filed on your behalf... https://github.com/GSA/terraform-kubernetes-aws-load-balancer-controller/issues/13
